fairlearn/postprocessing/_threshold_optimizer.py

```diff
@@ -211,6 +211,70 @@ class ThresholdOptimizer(BaseEstimator, MetaEstimatorMixin):
        Supervised Learning," arXiv.org, 07-Oct-2016.
        [Online]. Available: https://arxiv.org/abs/1610.02413.
 
+    Examples
+    --------
+    >>> from fairlearn.postprocessing import ThresholdOptimizer
+    >>> from fairlearn.metrics import MetricFrame, selection_rate
+    >>> from sklearn.metrics import accuracy_score
+    >>> from sklearn.linear_model import LogisticRegression
+    >>> from sklearn.model_selection import train_test_split
+    >>> from sklearn.preprocessing import StandardScaler
+    >>> from sklearn.pipeline import Pipeline
+    >>> from sklearn.metrics import balanced_accuracy_score
+    >>> import numpy as np
+
+    First we create some artificial data and fit an unmitigated pipeline to it
+
+    >>> X                  = np.array([[0], [1], [2], [3], [4], [5], [6], [7], [8], [9]])
```

Comment:
Don't get me wrong, your example is great! In fact, I would argue it's *too good* for the API docs. In the API docs it should be mostly about highlighting what you can pass to this particular API. I like [this example](https://scikit-learn.org/stable/modules/generated/sklearn.gaussian_process.GaussianProcessClassifier.html#sklearn.gaussian_process.GaussianProcessClassifier) from scikit-learn. They just set up the data, fit, predict, done. Things like train-test-split, preprocessing and pipelines, and evaluation with a `MetricFrame` are absolutely vital in a real example. In the API docs they're a bit too much IMO, though.

Adding @fairlearn/fairlearn-maintainers since we perhaps haven't done a great job at highlighting this.

What you've done might be a good fit for the `examples`. Another option you can consider is adding to the PR I started a while ago: #614 
There's also another related PR going on at the moment #997 just FYI

Reply:
No, it's actually by design. We use a package called sphinx-gallery for this. A more detailed explanation: https://fairlearn.org/main/contributor_guide/contributing_example_notebooks.html#formatting-of-example-notebooks